### PR TITLE
[TACHYON-1548] Combine BlockWorker and BlockDataManager

### PR DESCRIPTION
**Pull request by luoli523**
This patch will do the following:

* move the `BlockHeartbeatReporter` and `BlockMetricsReporter ` handler from `BlockDataManager` to `BlockWorker`
* Directly use the `BlockMasterClient` and `FileSystemMasterClient` in the `BlockWorker`
* initialize the `BlockStore` in the `BlockWorker`
* move the block process methods from `BlockDataManager` to `BlockWorker` using the `BlockStore` handler initialized directly in the `BlockWorker`
* use the `BlockWorker` `this` handler to initialize the `BlockMasterSync`,`PinListSync`,`SessionCleaner`,
`SpaceReserver` in `BlockWorker`
* delete `BlockDataManager` class
* update some other classes, using `BlockWorker` handler instead of `BlockDataManager`
* update some test case to use `BlockWorker` handler instead of `BlockDataManager`

### CODE REVIEW

**Comment by luoli523** [comment]
Change all the BlockWorker object name from mBlockDataManager to mBlockWorker.

File: keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorker.java [hunk 1]
```diff
@@ -51,13 +51,13 @@
    *
    * @param blockDataManager handler to the {@link BlockDataManager}
    */
-  public KeyValueWorker(BlockDataManager blockDataManager) {
+  public KeyValueWorker(BlockWorker blockDataManager) {
```
Comment (apc999) [inline keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorker.java]: change var name to `blockManager`?

File: keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorkerClientServiceHandler.java [hunk 2]
```diff
@@ -46,10 +46,10 @@
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   /** Block data manager for access block info */
-  private final BlockDataManager mBlockDataManager;
+  private final BlockWorker mBlockDataManager;
```
Comment (apc999) [inline keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorkerClientServiceHandler.java]: update member variable name ? also the javadoc

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java [hunk 3]
```diff
@@ -50,15 +50,15 @@
  */
 public final class WebInterfaceWorkerBlockInfoServlet extends HttpServlet {
   private static final long serialVersionUID = 4148506607369321012L;
-  private final transient BlockDataManager mBlockDataManager;
+  private final transient BlockWorker mBlockDataManager;
```
Comment (apc999) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: var name

File: servers/src/main/java/tachyon/web/WorkerUIWebServer.java [hunk 4]
```diff
@@ -41,16 +41,16 @@
    * @param conf Tachyon configuration
    */
   public WorkerUIWebServer(ServiceType serviceType, InetSocketAddress webAddress,
-      BlockDataManager blockdataManager, InetSocketAddress workerAddress, long startTimeMs,
+      BlockWorker blockWorker, InetSocketAddress workerAddress, long startTimeMs,
       TachyonConf conf) {
     super(serviceType, webAddress, conf);
-    Preconditions.checkNotNull(blockdataManager, "Block data manager cannot be null");
+    Preconditions.checkNotNull(blockWorker, "Block data manager cannot be null");
```
Comment (apc999) [inline servers/src/main/java/tachyon/web/WorkerUIWebServer.java]: update the message

File: servers/src/main/java/tachyon/worker/block/BlockMasterSync.java [hunk 5]
```diff
@@ -55,7 +55,7 @@
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
   private static final int DEFAULT_BLOCK_REMOVER_POOL_SIZE = 10;
   /** Block data manager responsible for interacting with Tachyon and UFS storage */
-  private final BlockDataManager mBlockDataManager;
+  private final BlockWorker mBlockWorker;
```
Comment (apc999) [inline servers/src/main/java/tachyon/worker/block/BlockMasterSync.java]: update the javadoc

File: servers/src/main/java/tachyon/worker/block/BlockMasterSync.java [hunk 6]
```diff
@@ -211,7 +211,7 @@ private void handleMasterCommand(Command cmd) throws Exception {
      * @param sessionId the session id
      * @param blockId the block id
      */
-    public BlockRemover(BlockDataManager blockDataManager, long sessionId, long blockId) {
+    public BlockRemover(BlockWorker blockDataManager, long sessionId, long blockId) {
```
Comment (apc999) [inline servers/src/main/java/tachyon/worker/block/BlockMasterSync.java]: varname

File: servers/src/main/java/tachyon/worker/nio/NIODataServer.java [hunk 7]
```diff
@@ -64,7 +64,7 @@
       Collections.synchronizedMap(new HashMap<SocketChannel, DataServerMessage>());
 
   // The block data manager.
-  private final BlockDataManager mDataManager;
+  private final BlockWorker mDataManager;
```
Comment (apc999) [inline servers/src/main/java/tachyon/worker/nio/NIODataServer.java]: update javadoc

File: servers/src/main/java/tachyon/worker/nio/NIODataServer.java [hunk 8]
```diff
@@ -77,12 +77,12 @@
    * @param dataManager the lock system for lock blocks
    * @param tachyonConf Tachyon configuration
    */
-  public NIODataServer(final InetSocketAddress address, final BlockDataManager dataManager,
+  public NIODataServer(final InetSocketAddress address, final BlockWorker blockWorker,
```
Comment (apc999) [inline servers/src/main/java/tachyon/worker/nio/NIODataServer.java]: javadoc is still `dataManager`

File: servers/src/test/java/tachyon/worker/file/FileDataManagerTest.java [hunk 9]
```diff
@@ -64,20 +64,20 @@ public void persistFileTest() throws Exception {
     List<Long> blockIds = Lists.newArrayList(1L, 2L);
 
     // mock block data manager
-    BlockDataManager blockDataManager = Mockito.mock(BlockDataManager.class);
+    BlockWorker blockWorker = Mockito.mock(BlockWorker.class);
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/file/FileDataManagerTest.java]: javadoc

File: servers/src/test/java/tachyon/worker/file/FileDataManagerTest.java [hunk 10]
```diff
@@ -139,7 +139,7 @@ public void errorHandlingTest() throws Exception {
     List<Long> blockIds = Lists.newArrayList(1L, 2L);
 
     // mock block data manager
-    BlockDataManager blockDataManager = Mockito.mock(BlockDataManager.class);
+    BlockWorker blockDataManager = Mockito.mock(BlockWorker.class);
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/file/FileDataManagerTest.java]: varname, comment

**Comment by apc999** [comment]
@luoli523  thanks for submitting the patch! The code structure is much better now!
One question: it seems `BlockDataManagerTest.java` is removed. Can we change it to `BlockWorkerTest`? If it is hard, you might want to comment all tests for now rather than removing the file.

I also left some minor comments regarding javadoc/comments/variable names. Thanks again for helping cleaning up the code!

**Comment by luoli523** [comment]
@apc999  Address the javadoc/varname/comments update.

Also, I renamed the BlockDataManagerTest -> BlockWorkerTest. But the cases in that ,I just comment them. I will finish the BlockWorkerTest in another issue because it will takes a little while and the master branch updated so quick, so now I just comment them for the quick merge.

Please check it again.

**Comment by luoli523** [comment]
The integration test case BlockWorkerClientAuthenticationIntegrationTest fail in a random rate because of other reason and have nothing to do with this update.

@apc999  
Please check this patch again.  before this patch merged, I have to resolve the conflict again and again...

**Comment by luoli523** [comment]
@apc999  any comments for this patch?  I created another issue to add the BlockWorkerTest unit test as TACHYON-1604

**Comment by apc999** [comment]
@luoli523  the PR looks good to me. I will merge this PR once it gets another LGTM.

File: keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorker.java [hunk 11]
```diff
@@ -42,7 +42,7 @@
   /** Configuration object */
   private final TachyonConf mTachyonConf;
   /** Block data manager for access block info */
```
Comment (caadecarvalho) [inline keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorker.java]: Shouldn't it refer to Block Worker now?

File: keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorker.java [hunk 12]
```diff
@@ -51,13 +51,13 @@
    *
    * @param blockDataManager handler to the {@link BlockDataManager}
    */
```
Comment (caadecarvalho) [inline keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorker.java]: Shouldn't it refer to Block Worker now?

File: keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorkerClientServiceHandler.java [hunk 13]
```diff
@@ -39,8 +39,8 @@
 import tachyon.thrift.KeyValueWorkerClientService;
 import tachyon.thrift.TachyonTException;
 import tachyon.thrift.ThriftIOException;
+import tachyon.worker.block.BlockWorker;
```
Comment (caadecarvalho) [inline keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorkerClientServiceHandler.java]: Shouldn't it be ordered?

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java [hunk 14]
```diff
@@ -50,15 +50,15 @@
  */
 public final class WebInterfaceWorkerBlockInfoServlet extends HttpServlet {
   private static final long serialVersionUID = 4148506607369321012L;
-  private final transient BlockDataManager mBlockDataManager;
+  private final transient BlockWorker mBlockWorker;
 
   /**
    * Creates a new instance of {@link WebInterfaceWorkerBlockInfoServlet}.
    *
    * @param blockDataManager block data manager
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: Shouldn't it refer to Block Worker now?

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerGeneralServlet.java [hunk 15]
```diff
@@ -145,9 +145,9 @@ public String getWorkerAddress() {
    * @param workerAddress worker address
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerGeneralServlet.java]: Shouldn't it mention blockWorker param?

File: servers/src/main/java/tachyon/worker/WorkerSource.java [hunk 16]
```diff
@@ -212,36 +212,36 @@ public void incBytesWrittenUfs(long n) {
    *
     * @param blockDataManager a block data manager handle
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/WorkerSource.java]: Shouldn't it refer to Block Worker now?

File: servers/src/main/java/tachyon/worker/block/BlockWorkerClientServiceHandler.java [hunk 17]
```diff
@@ -48,7 +48,7 @@
    *
    * @param worker block data manager handler
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/block/BlockWorkerClientServiceHandler.java]: Shouldn't it refer to Block Worker now?

File: servers/src/main/java/tachyon/worker/block/PinListSync.java [hunk 18]
```diff
@@ -33,7 +33,7 @@
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   /** Block data manager responsible for interacting with Tachyon and UFS storage */
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/block/PinListSync.java]: Shouldn't it refer to Block Worker now?

File: servers/src/main/java/tachyon/worker/block/SessionCleaner.java [hunk 19]
```diff
@@ -42,8 +42,8 @@
    *
    * @param blockDataManager a block data manager handle
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/block/SessionCleaner.java]: Shouldn't it refer to Block Worker now?

File: servers/src/main/java/tachyon/worker/block/SpaceReserver.java [hunk 20]
```diff
@@ -54,10 +54,10 @@
    *
    * @param blockManager a block manager handle
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/block/SpaceReserver.java]: Shouldn't it refer to Block Worker now?

File: servers/src/main/java/tachyon/worker/file/FileDataManager.java [hunk 21]
```diff
@@ -55,7 +55,7 @@
 
   private final UnderFileSystem mUfs;
   /** Block data manager for access block info */
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/file/FileDataManager.java]: same here.

File: servers/src/main/java/tachyon/worker/file/FileDataManager.java [hunk 22]
```diff
@@ -69,8 +69,8 @@
    *
    * @param blockDataManager a block data manager handle
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/file/FileDataManager.java]: same here.

File: servers/src/main/java/tachyon/worker/file/FileSystemWorker.java [hunk 23]
```diff
@@ -56,12 +56,12 @@
    * @param blockDataManager a block data manager handle
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/file/FileSystemWorker.java]: same here.

File: servers/src/main/java/tachyon/worker/netty/DataServerHandler.java [hunk 24]
```diff
@@ -70,8 +70,8 @@
    * @param dataManager a block data manager handle
```
Comment (caadecarvalho) [inline servers/src/main/java/tachyon/worker/netty/DataServerHandler.java]: same here.

**Comment by caadecarvalho** [comment]
Hi @luoli523 , i just left some comments regarding javadoc.

**Comment by luoli523** [comment]
@caadecarvalho  Thanks for the review. I addressed them and also fix some other javadoc issues.
Please confirm this again.
thank you

**Comment by apc999** [comment]
+1 LGTM

**Comment by yupeng9** [comment]
LGTM. Thanks for the refactoring.

**Comment by jsimsa** [comment]
FYI, the file https://github.com/amplab/tachyon/blob/master/core/server/src/main/java/tachyon/worker/package-info.java still contains references to BlockDataManager. This begs the question if we should be using package-info.java files (they are only used in two or three places).

**Pull request merged**